zingolib/src/commands/utils.rs

```diff
@@ -103,4 +118,172 @@ mod tests {
+                let result = parse_send_args(&args);
+                match result {
+                    Err(CommandError::MissingKey(e)) => assert_eq!(e, "amount".to_string()),
+                    _ => panic!(),
+                };
+            }
+            #[test]
+            fn non_string_address() {
+                let args = ["[{\"address\": 1, \"amount\": 123, \"memo\": \"testmemo\"}]"];
+                let result = parse_send_args(&args);
+                match result {
+                    Err(CommandError::UnexpectedType(e)) => {
+                        assert_eq!(e, "address not a Str!".to_string())
+                    }
+                    _ => panic!(),
+                };
+            }
+            #[test]
+            fn non_u64_amount() {
+                let args =
+                ["[{\"address\": \"testaddress\", \"amount\": \"Oscar Pepper\", \"memo\": \"testmemo\"}]"];
+                let result = parse_send_args(&args);
+                match result {
+                    Err(CommandError::NonJsonNumberForAmount(e)) => {
+                        assert_eq!(e, "amount is not a json::number::Number".to_string())
+                    }
+                    _ => panic!(),
+                };
+            }
+            #[test]
+            fn invalid_memo() {
+                let arg_contents =
+                    "[{\"address\": \"testaddress\", \"amount\": 123, \"memo\": \"testmemo\"}]";
+                let long_513_byte_memo = &"a".repeat(513);
+                let long_memo_args =
+                    arg_contents.replace("\"testmemo\"", &format!("\"{}\"", long_513_byte_memo));
+                let args = [long_memo_args.as_str()];
+
+                let result = parse_send_args(&args);
+                match result {
+                    Err(CommandError::InvalidMemo(e)) => {
+                        assert_eq!(
+                            e,
+                            format!(
+                                "Error creating output. Memo '\"{}\"' is too long",
+                                long_513_byte_memo.to_string()
+                            )
+                        )
+                    }
+                    _ => panic!(),
+                };
+            }
+        }
+        mod multi_string_args {
+            use super::*;
+            #[test]
+            fn two_args_wrong_amount() {
+                let args = ["testaddress", "foo"];
+                let result = parse_send_args(&args);
+                dbg!(&result);
+                match result {
+                    Err(CommandError::ParseIntFromString(e)) => {
+                        assert_eq!(
+                            "invalid digit found in string".to_string(),
+                            format!("{}", e)
+                        )
+                    }
+                    _ => panic!(),
+                };
+            }
+            #[test]
+            fn three_args_wrong_amount_show_trim_requirement() {
```

Comment:
as this is actually a test that is successful as oppose to verifying an error variant, can we move this to be a part of `parse_send_args_test` ?

Reply:
alternatively we could remove the `fail_parse_send_args` module and split `parse_send_args_test` into separate tests in its corresponding json or multilple_args module? so we just have two modules in the test module (json and multiple_args) containing both tests that pass and tests that show the correct error

Reply:
I generally like more smaller tests, but I also like the unit-centric pattern you're using, where there's a "this unit works right" test.

So..  I went with your option 1.